src/transaction.rs

```diff
@@ -0,0 +1,229 @@
+
+    fn seek<K>(&self, key: K) -> KvFuture<Scanner>
+    where
+        K: Into<Key>;
+
+    fn seek_reverse<K>(&self, key: K) -> KvFuture<Scanner>
+    where
+        K: Into<Key>;
+}
+
+pub trait Mutator {
+    fn set<P>(&mut self, pair: P) -> KvFuture<()>
+    where
+        P: Into<KvPair>;
+
+    fn delete<K>(&mut self, key: K) -> KvFuture<()>
+    where
+        K: Into<Key>;
+}
+
+pub struct Transaction;
+
+impl Transaction {
+    pub fn commit(&mut self) -> KvFuture<()> {
+        unimplemented!()
+    }
+
+    pub fn rollback(&mut self) -> KvFuture<()> {
+        unimplemented!()
+    }
+
+    pub fn lock_keys<I, K>(&mut self, keys: I) -> KvFuture<()>
+    where
+        I: IntoIterator<Item = I>,
+        K: Into<Key>,
+    {
+        drop(keys);
+        unimplemented!()
+    }
+
+    pub fn is_readonly(&self) -> bool {
+        unimplemented!()
+    }
+
+    pub fn start_ts(&self) -> Timestamp {
+        unimplemented!()
+    }
+
+    pub fn snapshot(&self) -> KvFuture<Snapshot> {
+        unimplemented!()
+    }
+}
+
+impl Retriever for Transaction {
+    fn get<K>(&self, key: K) -> KvFuture<Value>
+    where
+        K: Into<Key>,
+    {
+        drop(key);
+        unimplemented!()
+    }
+
+    fn batch_get<I, K>(&self, keys: I) -> KvFuture<Vec<KvPair>>
+    where
+        I: IntoIterator<Item = K>,
+        K: Into<Key>,
+    {
+        drop(keys);
+        unimplemented!()
+    }
+
+    fn seek<K>(&self, key: K) -> KvFuture<Scanner>
+    where
+        K: Into<Key>,
+    {
+        drop(key);
+        unimplemented!()
+    }
+
+    fn seek_reverse<K>(&self, key: K) -> KvFuture<Scanner>
+    where
+        K: Into<Key>,
+    {
+        drop(key);
+        unimplemented!()
+    }
+}
+
+impl Mutator for Transaction {
+    fn set<P>(&mut self, pair: P) -> KvFuture<()>
+    where
+        P: Into<KvPair>,
+    {
+        drop(pair);
+        unimplemented!()
+    }
+
+    fn delete<K>(&mut self, key: K) -> KvFuture<()>
+    where
+        K: Into<Key>,
+    {
+        drop(key);
+        unimplemented!()
+    }
+}
+
+pub struct Snapshot;
+
+impl Retriever for Snapshot {
+    fn get<K>(&self, key: K) -> KvFuture<Value>
+    where
+        K: Into<Key>,
+    {
+        drop(key);
+        unimplemented!()
+    }
+
+    fn batch_get<I, K>(&self, keys: I) -> KvFuture<Vec<KvPair>>
+    where
+        I: IntoIterator<Item = K>,
+        K: Into<Key>,
+    {
+        drop(keys);
+        unimplemented!()
+    }
+
+    fn seek<K>(&self, key: K) -> KvFuture<Scanner>
+    where
+        K: Into<Key>,
+    {
+        drop(key);
+        unimplemented!()
+    }
+
+    fn seek_reverse<K>(&self, key: K) -> KvFuture<Scanner>
+    where
+        K: Into<Key>,
+    {
+        drop(key);
+        unimplemented!()
+    }
+}
+
+pub struct Oracle;
+
+impl Oracle {
+    pub fn timestamp(&self) -> KvFuture<Timestamp> {
+        unimplemented!()
+    }
+
+    pub fn is_expired(&self, _lock_timestamp: Timestamp, _ttl: Timestamp) -> KvFuture<bool> {
```

Comment:
em, do we need this function?

Reply:
Not sure, just saw it exists in TiDB's code. I should read more code and figure out a reasonable API surface that users of TiKV expect. I might change it if it's not useful.

Reply:
Yes, you are right. Not only IsExpired, the whole Oracle thing is used internally and should not be exposed to public API. Let me remove it from public API.